### ISSUE TITLE
Path for enabling Debug mode in confg file incorrect

### ISSUE THREAD

**Issue by fiver-watson** [artefactual/atom-docs#126]
**Version**: 2.5

**Place in docs**: https://www.accesstomemory.org/docs/2.5/admin-manual/maintenance/debug/#using-qubit-dev-php

**Issue noted**

The path suggested for enabling Debug mode is currently listed as: 

```
/etc/php/7.2/fpm/php-fpm.conf
```

This is incorrect - the above file contains general PHP configuration values, nothing specific to AtoM. 

**Recommended fix**

The correct file and path in my Vagrant box is: 

```
/etc/php/7.2/fpm/pool.d/atom.conf
```

This file has a header noting it is an ansible managed file (which we use to automate deployments and the vagrant box creation at Artefactual) so we should confirm with a developer that this is also correct for manually installed tarball AtoM instances as well.

**Comment by rpearcemoses** [comment]
On my Vagrant instance (2.5), the file the recommended fix references doesn't exist. However, the last line in /etc/php/7.2/fpm/php-fpm.conf looks like it would call the file, if it existed. Not sure if adding another address (e.g., 10.10.10.10) to  env[ATOM_DEBUG_IP] = "127.0.0.1" would conflict with or be superceded by the value in php-fpm.conf. 

If /etc/php/7.2/fpm/pool.d/atom.conf is a good place to make this change, I recommend this wording:

A developer create or edit the file /etc/php/7.2/fpm/pool.d/atom.conf (called from /etc/php/7.2/fpm/php-fpm.conf) to add further IP addresses to the ATOM_DEBUG_IP variable. Doing so will allow you to access Debug mode from a different IP address than that of the server.

I've made these changes in a local dev fork and are ready to go when tested.

If it's better to edit the php-fpm.conf file, recommend adding a note to disregard the "do not edit" heading in pool.d/atom.conf.

**Comment by fiver-watson** [comment]
Hi @rpearcemoses , 

1) It's possible, since you had AtoM installed before, that you have an older box running PHP 7.0. can you check if the file exists at `/etc/php/7.0/fpm/pool.d/atom.conf` instead? I guess this would be the correct path for an Ubuntu 16.04 with PHP 7.0 installation, so we should probably clarify the difference for each version in our rewrite

You could also try checking the version of PHP you have installed by running `php -v`. 

2) I think this file **must** exist somewhere in your vagrant box, or else Nginx would likely be complaining. You could try the following command to see if you can find it: 

```
sudo find /etc/php/ -name atom.conf -ls
```

3) I don't think we should tell users that a developer could just create this file if it doesn't exist where expected. This file has a whole bunch of PHP configuration values that AtoM expects and needs. Creating a new file just to add the debug variable might have a lot of unintended consequences. 

4) In any case, I'm running into what may be a bug right now in testing this out. I can get the debug toolbar by adding `10.10.10.1` as my IP (which is what it typically is for the Vagrant box - note that this is different from the `.10` IP address we use to ACCESS the box. You can always check to see what IP the browser thinks you are coming from by looking at the IP included in the Nginx access log, at: `var/log/nginx/access.log`). However, when I try to log in while in Debug mode, it doesn't work. I'm going to follow up internally and see if someone else can reproduce this issue, and if so, I'll file a ticket for it. Will update you when I know more.

**Comment by rpearcemoses** [comment]
Aren't you supposed to be out of the office for a couple days? :-)

Just checked the instance at atom.pearcemoses.info. It's got php 7.0, and
there's a file at
/etc/php/7.0/fpm/pool.d/atom.conf

The Vagrant box is running 7.2, and it has the same file in the same
location (albeit 7.2).

It looks like the two files are virtually identical.  The file on the
Vagrant box begins with
;Ansible managed file, do not edit directly

The file on atom.pearcemoses.info does not have the line at the top.

Otherwise, the two files look virtually the same. The file on
atom.pearcemoses.info (7.0) has text at the bottom that matches the
illustration in the documentation beginning "This is a good place to define
some environment variables."  For whatever reason, this came with the
ATOM_DEBUG to include 10.10.10.10:
env[ATOM_DEBUG_IP] = "10.10.10.10,127.0.0.1"

The file on 7.2 Vagrant merely reads "Environment variables" and doesn't
include 10.10.10.10 as an option.
This file also includes php_admin_value[opcache.enable_cli] = 1  // missing
from 7.0

Based on the comparison, I'd be curious where the "Do not edit" comes from
in 7.2. Maybe because I configured the atom.pearcemoses.info manually? My
guess is that the manual could be tweaked to ignore that message and make
the change here?

I'm in over my head here.

-- R


-- Richard Pearce-Moses (personal account: pearcemoses@gmail.com)

**Comment by fiver-watson** [comment]
Path fixed in 2.6 docs in https://github.com/artefactual/atom-docs/commit/fafced498976d9cf9d8dd6eea96a0328f2c80282
